### DISCUSSION TITLE
RAD with C128 PAL issues

### DISCUSSION THREAD

**Discussion by Startaxxx** [frntc/RAD#50] [Q&A]
Hallo,
I’m experiencing problems using RAD with C128...
I tried with 2xC128's and 1xC128D - plastic (all PAL).
All of them show garbage chars around the screen. Background sound is ok.
Trying to mount any REU/Georam/Nuvies and pressing return to go back to basic, all 128s crash, so it’s practically impossibile to use. 
Tried changing the timing settings in the .cfg file without success.
I read the previous posts related to c128 in this conversation but honestly i didn't understand if anybody was able to solve the issue.
If yes, could somebody share the successful settings so i can try the same? Hope it will work.

By the way, no problems with the C64. Everything works smoothly with it.

Thank u very much for any help!!!

**Comment by insanity213** [comment]
Same here, never got it to work with the NTSC or PAL c128.  In fact, I tried it on my PAL c128 one more time yesterday and now the board appears to be dead even when putting back in the one C64 it actually worked in.  The C64 just hangs in reset and never boots.  Pretty sure I cooked something trying to get it to run on the 128.  Mines never been that stable in any machine though, pretty sure there's always been something wrong with it.

**Comment by Startaxxx** [comment]
Sorry to hear… 
As far as I understood very few people made it work, mostly with 128DCR’s…
I also tried with the dedicated PI power supply but maybe it’s even worse…
I think it simply doesn’t work with the 128, unless u’re lucky, and the risk is to fry everything by changing the configuration file. For me, no any success in total 4x128’s, different ASSY’s, ‘plain’, 
 charged with 64k video RAM and 128D. Nothing. 
I bought the RAD because it was written everywhere that it worked for 64/128, but it really seems it isn’t. Obviously, neither in 128’s 64 mode. 
Anyway, I really hope I’m wrong and would really appreciate somebody sharing effectively working settings. 
Or hopefully a patch to make it work with the 128 will be released soon…. There are very nice apps which could expand the 128+REU experience.

**Reply by frntc** [reply]
As written on Forum64.de:  try the timing adjustment in the menu (pound key), you'll get an immediate feedback when values become better/worse.

I never managed to fry any hardware, not even during development ;-)

**Comment by futureretrofusion** [comment]
My experience with the RAD is that you can even insert it upside down it
will not fry.  Stability can also be down to cold solder issues. I made a
board and it was playing up, but on closer inspection it was just cold
solder or missing solder. You could go over the board and reflow each
joint. Only problems I have heard about is people with NTSC flat 128s?

**Comment by Startaxxx** [comment]
I thought about mis-soldering, too, but it works like a charm with the c64… so I don’t really think soldering should be the guilty…
And obviously working with real-time adjustment via menu (£ key) is difficult because of crashes and un-readable and random chars…
That’s why I’m looking for working-PAL 128’s users who were successful by adjusting the timings so I can put the same values in the .cfg file and hopefully make it work and eventually fine-adjust. 
Really tricky, indeed…

**Comment by futureretrofusion** [comment]
Ahh sorry I misunderstood, I thought it wasn't working at all on 64 as
well. Hopefully there is a way to get it to stabilize on the 128.

Be good to see some more REU titles soon. All the purists need to accept
that Commodore put an expansion port for a reason, it has effectively
future proofed the c64 and the amiga and the others. It's the way they
intended, embrace the future :-)
can't wait to add AI to the c64 output ;-)

**Comment by Startaxxx** [comment]
…RAD can play a big part for the future of Commodore computers, to the point that Doom is the even the less interesting one, imho.
Honestly, RAD could also give to C128 the chances it didn’t have in the past, that’s why I’m really hoping in a firmware update and sharing experiences with eventual solutions…

**Comment by Nivlekk** [comment]
I'm using my RAD for over a year now but still not being able to get it working on my PAL C128 too. Works perfectly fine on both my C64s.
I've tried countless different timing combinations, but to no avail, unfortunately. I only managed to make the glitches go worse, never better.
Maybe it is due to certain revisions C128 or chip revisions, that have slightly different bus timings? I don't know... Quite clueless, actually.

Still hope that I also get it working with my C128 at some point.

**Reply by insanity213** [reply]
My theory is that due to the differences in how the 1Mhz clock (aka PHI2) is generated on the 128 vs the 64  makes finding correct timing harder on the 128, if not impossible.  The 64 has more delay on that line compared to the CPU clock input than the 128 does.  I have a 128 rigged up with a clock delay circuit to fix some SID timing issues and wanted to test the theory, but my RAD completely stopped working in any machine before I got the chance.  If I should come across another RAD at a decent price I'll try it but the only ones I've seen for sale near me recently are being sold my people, who IMHO, aren't abiding by the license agreement for this project.  It doesn't help that I live in the middle of nowhere.  Maybe someone else can run with it.  I would love to see this device work reliably on the 128 but can't work on it anymore due to my cart crapping out.

**Comment by futureretrofusion** [comment]
I can send you a rad for the cost of the postage .

**Reply by insanity213** [reply]
Oh wow, that'd be awesome!  I'm happy to pay a fair price too, just don't want to fall victim to a scammer who is trying to get rich on someone elses work.  Please send me an email so we can work through the details.  insanity -at- hellbent.cc

**Reply by insanity213** [reply]
Hey mate, got your package today.   Fired it up on a 64 that never wanted to work well with my original RAD and yours passed a REU RAM test without tweaking any values at all.  Thanks so much for this, I now have a known good specimin that I can continue to test with the 128.  You're awesome bro, thanks a million!

**Reply by Startaxxx** [reply]
Surely smarter than me!!!
Thank u very much for your work, buddy!
I’m sure u’ll succeed and u’ll climb the 8-bit Olympus.
Waiting for your updates!
Thank u

**Comment by futureretrofusion** [comment]
All good enjoy.

Both boards work, I always make them the same way, but at some point
swapped out the mouser parts for aliexpress parts, i think the larger board
has the mouser parts the more recent bbq special has the ali parts.
The only thing wrong with the larger board is that the gold came off the
edge contacts but there is nothing else wrong with it, both can work with
either zero2 or pi3.

I'm surprised that your board didn't work and stopped working, im sure its
cold solder somewhere, it could be undetectable even with a microscope,
just need to re flow all the joints with some fresh solder using a
soldering iron and some flux.
Issues with timing , and intermittent function, I've experienced that with
many boards that i made and then had to go over, i use magnification when i
work on these boards.
So in my experience the RAD can function in a limited way even when some
pins on some of the IC's are not making proper contact.

Let me know how you go.

All the best

**Reply by insanity213** [reply]
Hey mate, haven't forgotten about you.  I've gone about as far as I can down the PHI0/2 clock rabbit hole.  I've gotten the menu glitches down to a minimum (but still present) and memtest will reliably detect the proper amount of memory but the test never passes.  I think I'm going to shift my focus onto main memory timings of /CAS and maybe /RAS.  Starting to wonder if the corruption is happening because DMA writes from the REU to the system are happening when the main memory is not ready?  Just a wild ass guess based on what the screen corruption looks like.  I haven't given up yet! But this is slow and tedious work for me.  I'd appreciate any input from someone smarter than I am (which is basically anyone!).

**Reply by Startaxxx** [reply]
Thank u very much for ur efforts!!! I wasn’t successful in any ways unfortunately. 
Waiting for ur kind updates!!!

**Comment by mrfusion92** [comment]
Hi, I have too a C128D PAL with garbled screen. It's not only a graphical issue, when I try to load any reu image the system hangs. Even nuvies don't play. Doom worked and I could play it but still with the garbled screen. Sound seems to play fine, both menu and doom.

I tried adjusting the timings but nothing changed.

**Reply by insanity213** [reply]
You're not alone, this device doesn't work on the original motherboard in the 128D or the regular 128.  I suppose it could work on a 128DCR but I don't have one to test.  If the 128 is important to you, get a real REU or an Ultimate II cart.

**Reply by frntc** [reply]
@insanity213 the C128s are picky about the timings, but I actually developed the RAD using a flat/regular/original C128.

**Reply by mrfusion92** [reply]
@frntc so I just need to find the correct timings values?

**Reply by Startaxxx** [reply]
I tried almost all settings and it doesn’t work at all. 
please post yours if success. Honestly after a lot of time im giving up. 
At least 4x 128 and 128d pal

**Reply by lipteklips2k15** [reply]
> @insanity213 the C128s are picky about the timings, but I actually developed the RAD using a flat/regular/original C128.

can you please supply your timing settings - i cant get any joy on either my pal or ntsc flat c128 yet cart is fine in my breadbin and c64c. Also is it possible incorrect timing settijg can kill cia's? as since trying the reu on both my machines they now have missing keys!

**Comment by lipteklips2k15** [comment]
peice of shit has now killed my last working c128.
